cmd/gapid/main.go

```diff
@@ -0,0 +1,225 @@
+		fmt.Println(" --console         Run GAPID inside a terminal console")
+	}
+
+	if console && runtime.GOOS == "windows" {
+		fmt.Println()
+		fmt.Println("Press enter to continue")
+		os.Stdin.Read(make([]byte, 1))
+	}
+
+	if failed {
+		os.Exit(1)
+	}
+}
+
+func locateVM(cwd, vm string, console bool) (string, bool) {
+	if vm != "" {
+		if checkVM(vm, false) {
+			return vm, false
+		}
+		if java := javaInHome(vm, console); checkVM(java, false) {
+			return java, false
+		}
+		fmt.Printf("JVM \"%s\" not found.\n", vm)
+		return "", true
+	}
+
+	if java := javaInHome(filepath.Join(cwd, "jre"), console); checkVM(java, true) {
+		return java, false
+	}
+
+	binary := "java"
+	if runtime.GOOS == "windows" && !console {
+		binary = "javaw"
+	}
+	if fromPath, err := exec.LookPath(binary); err != nil && checkVM(fromPath, true) {
+		return fromPath, false
+	}
+
+	if home := os.Getenv("JAVA_HOME"); home != "" {
+		if java := javaInHome(home, console); checkVM(java, true) {
+			return java, false
+		}
+	}
+
+	if runtime.GOOS == "linux" {
+		if java := "/usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java"; checkVM(java, true) {
+			return java, false
+		}
+	}
+
+	fmt.Println("No suitable JVM found.")
+	return "", true
+}
+
+func javaInHome(home string, console bool) string {
+	java := filepath.Join(home, "bin", "java")
+	if runtime.GOOS == "windows" {
+		if console {
+			java += ".exe"
+		} else {
+			java += "w.exe"
+		}
+	}
+
+	return java
+}
+
+func checkVM(java string, checkVersion bool) bool {
+	if stat, err := os.Stat(java); err != nil || stat.IsDir() {
+		return false
+	}
+
+	if !checkVersion {
+		return true
+	}
+
+	version, err := exec.Command(java, "-version").CombinedOutput()
+	if err != nil {
+		return false
+	}
+
+	// Looks for the pattern: <product> version "<major>.<minor>.<micro><build>"
+	// Not using regular expressions to avoid binary bloat.
+	versionStr := string(version)
+	if p := strings.Index(versionStr, versionPrefix); p >= 0 {
+		p += len(versionPrefix)
+		if q := strings.Index(versionStr[p:], "."); q > 0 {
+			if r := strings.Index(versionStr[p+q+1:], "."); r > 0 {
+				major, _ := strconv.Atoi(versionStr[p : p+q])
+				minor, _ := strconv.Atoi(versionStr[p+q+1 : p+q+r+1])
+				return major > 1 || (major == 1 && minor >= 8)
+			}
+		}
+	}
+
+	return false
+}
+
+func locateGapic(cwd string) (string, bool) {
+	gapic := filepath.Join(cwd, "lib", "gapic.jar")
+	if _, err := os.Stat(gapic); !os.IsNotExist(err) {
+		return gapic, false
+	}
+
+	var buildGapic string
+	switch runtime.GOOS {
+	case "linux", "windows":
+		buildGapic = filepath.Join(cwd, "..", "java", "gapic-"+runtime.GOOS+".jar")
+	case "darwin":
+		buildGapic = filepath.Join(cwd, "..", "java", "gapic-osx.jar")
+	default:
+		buildGapic = filepath.Join(cwd, "..", "java", "gapic.jar")
+	}
+
+	if _, err := os.Stat(buildGapic); !os.IsNotExist(err) {
+		return buildGapic, false
+	}
+
+	fmt.Printf("GAPIC JAR \"%s\" not found.\n", gapic)
```

Comment:
Protip: Use backticks to avoid nasty escaping:

```go
fmt.Printf(`GAPIC JAR "%s" not found.\n`, gapic)
```

Or do what I typically do, which is use single quotes:

```go
fmt.Printf("GAPIC JAR '%s' not found.\n", gapic)
```